atsd_webtests/src/test/java/com/axibase/webtest/cases/MetricSearchTest.java

```
@@ -1,75 +1,83 @@
+                        "metricsearchtest_contains_underscore"},
+                {".", "metricsearchtest_contains-hyphen1", "metricsearchtest_contains-hyphen2",
+                        "metricsearchtest_contains.point1", "metricsearchtest_contains.point2", "metricsearchtest_contains.point3",
+                        "metricsearchtest_contains_underscore"},
+                {"metricsearchtest", "metricsearchtest_contains-hyphen1", "metricsearchtest_contains-hyphen2",
+                        "metricsearchtest_contains.point1", "metricsearchtest_contains.point2", "metricsearchtest_contains.point3",
+                        "metricsearchtest_contains_underscore"},
+                {" metricsearchtest ", "metricsearchtest_contains-hyphen1", "metricsearchtest_contains-hyphen2",
+                        "metricsearchtest_contains.point1", "metricsearchtest_contains.point2", "metricsearchtest_contains.point3",
+                        "metricsearchtest_contains_underscore"},
+                {"some-bad-mask", new String[]{}},
+                {"cd s", new String[]{}}
+        };
     }
 
     @BeforeClass
     public void generateData() {
         super.setUp();
-        open("/metrics/entry");
-        CommonActions.sendTextToCodeMirror($(By.name("commands")), "<#list 1..2 as i> \n" +
-                "series e:metric-for-entity-search_entity m:contains-hyphen${i}=${i} \n" +
-                "</#list>\n");
-        CommonActions.sendTextToCodeMirror($(By.name("commands")), "<#list 1..3 as i> \n" +
-                "series e:metric-for-entity-search_entity m:contains.point${i}=${i} \n" +
-                "</#list>\n");
-        CommonActions.sendTextToCodeMirror($(By.name("commands")), "" +
-                "series e:metric-for-entity-search_entity m:contains_underscore=1 \n");
-        $("button[value=send]").click();
+        DataEntryPage dataEntryPage = new DataEntryPage();
```

Lines 55-66 can be a single method to type, sumbit and test insertion is successfully.

---

Also it will be better to throw special exception for invalid commands, with message received from client error box